src/bun.js/api/bun/subprocess.zig

```diff
@@ -2156,9 +2156,15 @@ pub fn spawnMaybeSync(
                 }
             }
 
-            if (try args.get(globalThis, "timeout")) |val| {
-                if (val.isNumber() and val.isFinite()) {
-                    timeout = @max(val.coerce(i32, globalThis), 1);
+            if (try args.get(globalThis, "timeout")) |timeout_value| brk: {
```

Comment:
We were incorrectly treating a timeout of 0 as being a timeout of 1. We shouldn't do that.